### ISSUE TITLE
SymmetricTensor follow-ups

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#21269]
## Reason
Once the SymmetricTensor PR is merged there are some things to add in a follow-up PR.

- Add `transposeTime` (`A^T*A`)
- Add `SymmetricRankTwoTensor::doubleContrction(RankTwoTensor)`

## Design
New functionality will mirror existing one closely.

## Impact
Added capability.
